### PR TITLE
feat: Add i18next documentation

### PR DESCRIPTION
**Pull request by stepan662**

### CODE REVIEW

File: docs/web/using_with_i18next/installation.mdx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  apiKey: ...,
+  ui: ...,
+  ...
+})
+  .use(...)
+  .init(...)
+
+```
+
+This way `Tolgee` will serve as an [backend](https://www.i18next.com/misc/creating-own-plugins#backend) and will react to language changes. `withTolgee` also adds custom [post processor](https://www.i18next.com/misc/creating-own-plugins#post-processor) for translation wrapping. Also it manipulates the config to work properly, however you can still configurate `i18next` the way you want, changes to config are purely additive and your config will be carefully extended.
```
[the remark this answers is not in the document]
Comment (JanCizmar) [reply docs/web/using_with_i18next/installation.mdx]: to work properly, however you ca
->
to work properly. However, you

[the remark this answers is not in the document]
Comment (JanCizmar) [reply docs/web/using_with_i18next/installation.mdx]: the way you want. Changes to config

File: docs/web/using_with_i18next/api.mdx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+### function `withTolgee(i18next, tolgeeConfig)`
+
+Ensures complete integration between i18next and tolgee.
+
+- `i18next` - i18next instance
+- `tolgeeConfig` - [`TolgeeConfiguration`](../configuration) type (wrapperMode is automatically set to `invisible` if not defined)
+
+## Custom integration
+
+If you need more granual control of what is applied to i18next instance, you can use following functions manually instead of using `withTolgee`, which uses them all.
```
Comment (marketachalupnikova) [inline docs/web/using_with_i18next/api.mdx:19]: granular

File: docs/web/using_with_i18next/api.mdx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+### function `tolgeeBackend(tolgee)`
+
+Creates tolgee [backend](https://www.i18next.com/misc/creating-own-plugins#backend) plugin for i18next.
```
Comment (marketachalupnikova) [inline docs/web/using_with_i18next/api.mdx:23]: Tolgee

File: docs/web/using_with_i18next/api.mdx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+- `tolgee` - Tolgee instance
+
+### function `tolgeeProcessor(tolgee)`
+
+Creates tolgee [postProcessor](https://www.i18next.com/misc/creating-own-plugins#post-processor) plugin for i18next.
```
Comment (marketachalupnikova) [inline docs/web/using_with_i18next/api.mdx:29]: Tolgee

File: docs/web/using_with_i18next/api.mdx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+Applies wrapping to each translation, so it's detectable for in-context in dev mode.
+
+- `tolgee` - Tolgee instance
+
+### function `tolgeeApply(tolgee, i18next)`
+
+Registers necessary callbacks (`tolgee.onTranslationChange` and `i18next.on('languageChanged')`) on tolgee instance and on i18next.
```
Comment (marketachalupnikova) [inline docs/web/using_with_i18next/api.mdx:36]: Tolgee

File: docs/web/using_with_i18next/api.mdx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+- `tolgee` - Tolgee instance
+- `i18next` - i18next instance
+
+### function `tolgeeOptions(options)`
+
+Extends i18next's [`InitOptions`](https://www.i18next.com/overview/configuration-options) with options that are necessary for tolgee to work properly. We need to globally enable our `tolgeeProcessor` and also react on store event `added`, so the ui is refreshed when some translation is changed through in-context edit.
```
Comment (marketachalupnikova) [inline docs/web/using_with_i18next/api.mdx:43]: Tolgee

**Review by JanCizmar — APPROVED** [review]

File: docs/web/using_with_i18next/api.mdx (hunk 6 above)
Comment (marketachalupnikova) [reply docs/web/using_with_i18next/api.mdx:43]: UI

File: docs/web/using_with_i18next/react_integration.mdx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+title: React integration (i18next)
+sidebar_label: React integration
+slug: /web/using_with_i18next/react_integration
+---
+
+:::info
+To get full image of working i18next integration, check our [React i18next Example Application](https://github.com/tolgee/react-i18next-example).
+:::
+
+For integration with react use standard react extension + we recommend also use ICU message formatter as that is supported on tolgee platform natively:
```
Comment (marketachalupnikova) [inline docs/web/using_with_i18next/react_integration.mdx:12]: React

Comment (marketachalupnikova) [reply docs/web/using_with_i18next/react_integration.mdx:12]: Tolgee

File: docs/web/using_with_i18next/api.mdx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+- `tolgee` - Tolgee instance
+
+### function `tolgeeProcessor(tolgee)`
+
+Creates tolgee [postProcessor](https://www.i18next.com/misc/creating-own-plugins#post-processor) plugin for i18next.
+Applies wrapping to each translation, so it's detectable for in-context in dev mode.
+
+- `tolgee` - Tolgee instance
+
+### function `tolgeeApply(tolgee, i18next)`
```
[the remark this answers is not in the document]
Comment (stepan662) [reply docs/web/using_with_i18next/api.mdx:34]: This way it's in line with tolgeeBackend and tolgeeProcessor

[the remark this answers is not in the document]
Comment (JanCizmar) [reply docs/web/using_with_i18next/api.mdx:34]: OK

**Review by JanCizmar — APPROVED** [review]

**Pull request merged**